### PR TITLE
zio: Introduce hts221 humidity sensor

### PR DESCRIPTION
**Pull request by avisconti**
depends on #16119
Prepare HTS221 driver based on zio interface.
This sensor component does not have an internal hardware FIFO. Nevertheless
I defined a s/w fifo of length==1. I will try later on with other sensors with h/w fifo.

In order to test it on the x_nucleo_shield3 is rebased on master and not topic-sensors.
I can later switch it if needed

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]

File: boards/shields/x_nucleo_iks01a3/x_nucleo_iks01a3.overlay [hunk 1]
```diff
@@ -6,11 +6,14 @@
 
 &arduino_i2c {
 
+#if 1
```
Comment (ulfalizer) [inline boards/shields/x_nucleo_iks01a3/x_nucleo_iks01a3.overlay]: Leftover debugging?

File: boards/shields/x_nucleo_iks01a3/x_nucleo_iks01a3.overlay [hunk 2]
```diff
@@ -39,3 +42,17 @@
 		label = "LSM6DSO";
 	};
 };
+&arduino_spi {
+	cs-gpios = <&gpiob 6 0>;
+
+#if 0
```
Comment (ulfalizer) [inline boards/shields/x_nucleo_iks01a3/x_nucleo_iks01a3.overlay]: Leftover debugging?

File: drivers/Kconfig [hunk 3]
```diff
@@ -89,4 +89,7 @@ source "drivers/neural_net/Kconfig"
 
 source "drivers/hwinfo/Kconfig"
 
+source "drivers/zio/Kconfig"
+
+
```
Comment (ulfalizer) [inline drivers/Kconfig]: Accidental extra blank line.

File: drivers/sensor/CMakeLists.txt [hunk 4]
```diff
@@ -23,7 +23,8 @@ add_subdirectory(grove)
 add_subdirectory_ifdef(CONFIG_TI_HDC		ti_hdc)
 add_subdirectory_ifdef(CONFIG_HMC5883L		hmc5883l)
 add_subdirectory_ifdef(CONFIG_HP206C		hp206c)
-add_subdirectory_ifdef(CONFIG_HTS221		hts221)
+# test hts221 from zio
+#add_subdirectory_ifdef(CONFIG_HTS221		hts221)
```
Comment (ulfalizer) [inline drivers/sensor/CMakeLists.txt]: Leftover debugging?

File: drivers/sensor/Kconfig [hunk 5]
```diff
@@ -71,7 +71,8 @@ source "drivers/sensor/hmc5883l/Kconfig"
 
 source "drivers/sensor/hp206c/Kconfig"
 
-source "drivers/sensor/hts221/Kconfig"
+# test hts221 from zio
+#source "drivers/sensor/hts221/Kconfig"
```
Comment (ulfalizer) [inline drivers/sensor/Kconfig]: Leftover debugging?

File: dts/bindings/sensor/focaltech,ft5336.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) 2019, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
```
Comment (ulfalizer) [inline dts/bindings/sensor/focaltech,ft5336.yaml]: Redundant document separator (see https://github.com/zephyrproject-rtos/zephyr/pull/16913).

File: dts/bindings/sensor/focaltech,ft5336.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+properties:
+    compatible:
+      constraint: "focaltech,ft5336"
+
+    int-gpios:
+      type: compound
+      category: optional
+      generation: define, use-prop-name
+...
```
Comment (ulfalizer) [inline dts/bindings/sensor/focaltech,ft5336.yaml]: Redundant document separator (see https://github.com/zephyrproject-rtos/zephyr/pull/16913).

**Comment by bfrog** [comment]
@avisconti Glad to see another driver being written against this API, the current API I believe needs more refinement and I began working on that on a PR.

I would love feedback on the API as it stands here from you as you've now written a driver for it.

I'd also love to get feedback on what I think is a simpler version of this idea in this pull request
https://github.com/zephyrproject-rtos/zephyr/pull/17434

I think topic sensors needs to be rebased anyways, I'll see if I can do it or if I can find someone to do it for me. Then I can take a closer look at this

**Comment by carlescufi** [comment]
@avisconti `zio` has been replaced with `rtio`. #16119 will be rewritten to target `rtio`, and once that is done we need to do the same with this PR.

**Comment by avisconti** [comment]
> @avisconti `zio` has been replaced with `rtio`. #16119 will be rewritten to target `rtio`, and once that is done we need to do the same with this PR.

Yes, sure. Thanks!
I'll constantly catch up with latest news.

**Pull request closed without merging**